kurl_proxy/cmd/main.go

```diff
@@ -86,16 +83,18 @@ func main() {
 	if err != nil {
 		log.Panic(err)
 	}
+	// TODO: Assert namespace not empty, else we get secrets from all namespaces
 	secrets := clientset.CoreV1().Secrets(namespace)
 
 	_, err = secrets.Get(context.Background(), tlsSecretName, metav1.GetOptions{})
 	if err != nil {
-		log.Print("creating tls secret")
+		log.Print("creating default tls secret")
 
 		err = generateDefaultCertSecret(secrets, namespace)
 		if err != nil {
 			log.Printf("Could not regenerate default certificate: %v", err)
 		}
+		// TODO: Why are we exiting here? It leads to a pod restart
```

Comment:
not sure. on the surface, it seems like we dont _need_ a restart for anything

Reply:
I'll leave that behaviour for now, but leave the TODO there so as not to introduce some unwanted behaviour change.